File: CyborgPunch/CyborgPunch/CyborgPunch/Core/Component.cs
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;

namespace CyborgPunch.Core
{
    public abstract class Component
    {
        public bool enabled = true;
        public Blob blob;

        protected Component()
        {
        }

        public virtual void Update()
        {
        }

        public virtual void Draw(SpriteBatch spriteBatch)
        {
        }

        public virtual void DrawDebug(SpriteBatch spriteBatch)
        {
        }

        public virtual void Start()
        {
        }
    }
}

File: PunchHarder/trunk/PunchHarder/engine/component.cpp
#include "engine.h"

Component::Component()
{
    m_pGameObject = NULL;
}

Component::~Component()
{
    Debug::Log("Destructor Component");
}

void Component::Start()
{
}

void Component::OnDestroy()
{
}

void Component::Update()
{
}

void Component::Draw()
{
}
